service/instance.py

```diff
@@ -1521,13 +1522,25 @@ def _to_user_driver(core_identity):
         (auth, sess, token) = _token_to_keystone_scoped_project(
             auth_url, auth_token,
             project_name, domain_name)
+        user_driver = UserManager(auth_url=auth_url,
```

Comment:
Actually it is more than just covering bases.  When libcloud object is constructed using a token, the constructor requires and sets the token, the auth_url and the base_url.  There is no setter for any of the these three values, so there is no way to change after the object is constructed.  This has the advantage that if lib cloud exists, it has stored everything that it needs internally to make calls for as long as the token is valid.  The disadvantage is that you need to authenticate every time you create the object (or switch projects or ...).

The session does include the token and a couple of methods to get the token and the associated project.  I don't think it has the auth_url or base url - both of these are in the catalog.

There are better designs to handle this, but all of them would involve rewriting rtwo.